### ISSUE TITLE
[BUG] Loading Save Not Resetting $Data

### ISSUE THREAD

**Issue by PinkPenguin418** [liana-p/narrat-engine#255]
**Describe the bug**
When you load a previous save, it does not reset the data object to previous state, it keeps current state.

**To Reproduce**
1. set data.someNumber 2
2. Manually Save
3. set data.someNumber 3
4. jump to some label
5. Load Save
6. display $data.someNumber
7. Will show data.someNumber as 3

**Expected behavior**
I'd expect the data.someNumber to be a 2

**Screenshots**

**Browser**
Chrome, latest version

**Comment by PinkPenguin418** [comment]
So I noticed a few things while digging into the code. Not sure if it's related but I saw that in game-save.ts the GlobalGameSave has data there as well as achievements. That feels odd as I expected to find "global" instead of "data" (per the documentation in save-and-load.md. 

I tried changing that, but it didn't fix the problem, so I'm guessing more is at work here.

**Comment by PinkPenguin418** [comment]
So I did a lot more digging on this. What I've noticed is this situation only happens after you loaded a game. It would appear that load doesn't load a copy of the vm store from the save file but actually uses that vm from the save file. Which means any changes to the game after loading the file results to changes to the underlying save file. Obviously that's not great.

I eventually found the culprit in the code. The loadSaveData is passing data.data as a reference. Simply needed to add deepCopy().
